woocommerce/woocommerce-ios
# Added ResponseType extension to SiteSettingMapper

Closes: #1201 

### Description

When we decode the site settings API response into our SiteSettings model, we [assume](https://github.com/woocommerce/woocommerce-ios/blob/81ed7b18078c58d3337f53a91d0f09c802bf3c3f/Networking/Networking/Model/SiteSetting.swift#L44) that all values we're interested in will be strings, however,  these could be a mixed type. This causes that properties like `woocommerce_all_except_countries`, `woocommerce_specific_allowed_countries`, and `woocommerce_specific_ship_to_countries` cannot be decoded properly, as their value is an array, and warnings are logged into the application as a result.

This PR addresses handling the warning by checking the response type, and filtering these by string (supported) or non-string (non-supported yet) values.

### Testing instructions
- Run the app
- Check in the debugger for any of the following warnings:
  - Could not successfully decode SiteSetting value for woocommerce_all_except_countries
  - Could not successfully decode SiteSetting value for woocommerce_specific_allowed_countries
  - Could not successfully decode SiteSetting value for woocommerce_specific_ship_to_countries
- Note that these should not appear.

### Screenshots


---
